internal/infrastructure/manager.go

```diff
@@ -36,14 +36,19 @@ type Manager interface {
 // NewManager returns a new infrastructure Manager.
 func NewManager(cfg *config.Server) (Manager, error) {
 	var mgr Manager
-	if cfg.EnvoyGateway.Provider.Type == egv1a1.ProviderTypeKubernetes {
+
+	switch cfg.EnvoyGateway.Provider.Type {
+	case egv1a1.ProviderTypeKubernetes:
 		cli, err := client.New(clicfg.GetConfigOrDie(), client.Options{Scheme: envoygateway.GetScheme()})
 		if err != nil {
 			return nil, err
 		}
 		mgr = kubernetes.NewInfra(cli, cfg)
-	} else {
-		// Kube is the only supported provider type for now.
+
+	case egv1a1.ProviderTypeFile:
```

Comment:
i dont think this is a valid case - `file` is a valid resource provider, not infra provider

Reply:
this is a old api. updated